### PR TITLE
feat(api/conf) add FindHostSeverity and UpdateHostSeverity, update AddHostSeverity

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New API endpoints  :
``` 
GET /configuration/hosts/severities/{id}
PUT /configuration/hosts/severities/{id}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)
- [x] 23.10.0

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/HostSeverity/Application/UseCase/FindHostSeverity/FindHostSeverity.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    /**
+     * @param HostSeverity $hostSeverity
+     *
+     * @return FindHostSeverityResponse
+     */
+    private function createResponse(
+        HostSeverity $hostSeverity
+    ): FindHostSeverityResponse {
```
Comment (jeremyjaouen) [inline centreon/src/Core/HostSeverity/Application/UseCase/FindHostSeverity/FindHostSeverity.php:117]:
```suggestion
    private function createResponse(HostSeverity $hostSeverity): FindHostSeverityResponse
    {
```

File: centreon/src/Core/HostSeverity/Domain/Model/HostSeverity.php [hunk 2]
```diff
@@ -53,4 +54,47 @@ public function getId(): int
     {
         return $this->id;
     }
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void
```
Comment (callapa) [inline centreon/src/Core/HostSeverity/Domain/Model/HostSeverity.php:61]: The exceptions of the assertions are missing.

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php [hunk 3]
```diff
@@ -67,7 +67,7 @@ public function __invoke(AddHostSeverityRequest $request, PresenterInterface $pr
                     ['user_id' => $this->user->getId()]
                 );
                 $presenter->setResponseStatus(
-                    new ForbiddenResponse(HostSeverityException::addNotAllowed()->getMessage())
+                    new ForbiddenResponse(HostSeverityException::writeActionsNotAllowed()->getMessage())
```
Comment (abuathier) [inline centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php:70]:
```suggestion
                    new ForbiddenResponse(HostSeverityException::writeActionsNotAllowed())
```

File: centreon/src/Core/HostSeverity/Application/UseCase/UpdateHostSeverity/UpdateHostSeverity.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    public function __invoke(UpdateHostSeverityRequest $request, PresenterInterface $presenter): void
+    {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $this->error(
+                    "User doesn't have sufficient rights to update host severities",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostSeverityException::writeActionsNotAllowed()->getMessage())
```
Comment (abuathier) [inline centreon/src/Core/HostSeverity/Application/UseCase/UpdateHostSeverity/UpdateHostSeverity.php:71]:
```suggestion
                    new ForbiddenResponse(HostSeverityException::writeActionsNotAllowed())
```

**Review by abuathier — DISMISSED** [review]

**Review by abuathier — APPROVED** [review]

File: centreon/tests/php/Core/HostSeverity/Application/UseCase/UpdateHostSeverity/UpdateHostSeverityTest.php [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        ->willReturn(false);
+
+    $this->request->alias = '';
+
+    ($this->useCase)($this->request, $this->presenter);
+
+    expect($this->presenter->getResponseStatus())->toBeInstanceOf(InvalidArgumentResponse::class);
+});
+
+it('should return NoContentResponse on success', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Application/UseCase/UpdateHostSeverity/UpdateHostSeverityTest.php:236]:
```suggestion
it('should return a NoContentResponse on success', function () {
```

File: centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php [hunk 6]
```diff
@@ -135,3 +135,100 @@
         ->getMessage()
 );
 
+it('should throw an exception when host severity name is set empty', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php:138]:
```suggestion
it('should throw an exception when host severity name is empty', function () {
```

File: centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php [hunk 7]
```diff
@@ -135,3 +135,100 @@
...
+        $this->iconId
+    );
+    $hostSeverity->setName('');
+})->throws(
+    \Assert\InvalidArgumentException::class,
+    AssertionException::notEmptyString('HostSeverity::name')
+        ->getMessage()
+);
+
+it('should throw an exception when host severity name is set too long', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php:153]:
```suggestion
it('should throw an exception when host severity name is too long', function () {
```

File: centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php [hunk 8]
```diff
@@ -135,3 +135,100 @@
...
+    \Assert\InvalidArgumentException::class,
+    AssertionException::maxLength(
+        str_repeat('a', HostSeverity::MAX_NAME_LENGTH + 1),
+        HostSeverity::MAX_NAME_LENGTH + 1,
+        HostSeverity::MAX_NAME_LENGTH,
+        'HostSeverity::name'
+    )->getMessage()
+);
+
+it('should throw an exception when host severity alias is set empty', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php:172]:
```suggestion
it('should throw an exception when host severity alias is empty', function () {
```

File: centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php [hunk 9]
```diff
@@ -135,3 +135,100 @@
...
+        $this->iconId
+    );
+    $hostSeverity->setAlias('');
+})->throws(
+    \Assert\InvalidArgumentException::class,
+    AssertionException::notEmptyString('HostSeverity::alias')
+        ->getMessage()
+);
+
+it('should throw an exception when host severity alias is set too long', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php:187]:
```suggestion
it('should throw an exception when host severity alias is too long', function () {
```

File: centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php [hunk 10]
```diff
@@ -135,3 +135,100 @@
...
+    \Assert\InvalidArgumentException::class,
+    AssertionException::maxLength(
+        str_repeat('a', HostSeverity::MAX_ALIAS_LENGTH + 1),
+        HostSeverity::MAX_ALIAS_LENGTH + 1,
+        HostSeverity::MAX_ALIAS_LENGTH,
+        'HostSeverity::alias'
+    )->getMessage()
+);
+
+it('should throw an exception when host severity level is set too high', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Domain/Model/HostSeverityTest.php:206]:
```suggestion
it('should throw an exception when host severity level is too high', function () {
```

**Review by cg-tw — DISMISSED** [review]

**Review by abuathier — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**